### ISSUE TITLE
Add a new capability for generarting these reports

### ISSUE THREAD

**Issue by jpahullo** [marcusgreen/moodle-report_advancedgrading#40]
**Describe the need**
We have detected that to generate these reports, user needs to be in a role with the capability `moodle/grade:edit`.

In certain circumstances, it is not desirable: external reviewers to institution need to check how the evaluation was to all students. Granting this capability to these users allow them also to modify existing grades.

**Expected behavior**
So, the proposal is to add a separate capability from the own plugin, which only certain roles may have granted, without the security they will not modify exisiting grades.

**Comment by marcusgreen** [comment]
That sounds like a very good idea

**Comment by jpahullo** [comment]
Hi @marcusgreen ,

Would you be interested in receiving a contribution in this direction? In our use case, as exposed, and after having evaluted other possibilities, we believe that this is the best option for us.

Let us know.

Thanks,

Jordi

**Comment by marcusgreen** [comment]
Yes definitly. I have been distracted by other things (such as AI) over the christmas period. Let me know more about your ideas.

**Comment by jpahullo** [comment]
It sounds really enjoying these things about AI. hehehe

The main idea we have is broadly speaking the following:

1. Adding a new capability into the plugin with the meaning "allow the user to see the rubric detail", with a name like: `report/advancedgrading:view`, into the `db/access.php`
2. Use this new capability throughout the plugin, instead of the `moodle/grade:edit`.

In general, these would be the main changes that would include the patch.

Do you like the proposal?

Thanks!

Jordi

**Comment by marcusgreen** [comment]
I like the proposal a lot

**Comment by jpahullo** [comment]
@marcusgreen : new PR #46 opened, poiniting to main branch. Hopefully the CI will pass on green.
